pkg/client/typed_client.go

```diff
@@ -143,6 +144,68 @@ func (c *typedClient) Get(ctx context.Context, key ObjectKey, obj Object) error
 		Name(key.Name).Do(ctx).Into(obj)
 }
 
+func (c *typedClient) ListPages(ctx context.Context, obj ObjectList,
+	callback func(obj ObjectList) error, opts ...ListOption) error {
+	r, err := c.cache.getResource(obj)
+	if err != nil {
+		return err
+	}
+	listOpts := ListOptions{}
+	listOpts.ApplyOptions(opts)
+
+	// Fetch items at chunks of one hundred if not specified differently.
+	if listOpts.Limit == 0 {
+		Limit(DefaultPageLimit).ApplyToList(&listOpts)
+	}
+
+	// Retrieve initial chunck of data.
+	var allItems []runtime.Object
+	var interimResult ObjectList
+	err = r.Get().
+		NamespaceIfScoped(listOpts.Namespace, r.isNamespaced()).
+		Resource(r.resource()).
+		VersionedParams(listOpts.AsListOptions(), c.paramCodec).
+		Do(ctx).Into(interimResult)
+	if err != nil {
+		return err
+	}
+
+	if err := callback(interimResult); err != nil {
+		return err
+	}
+
+	items, err := apimeta.ExtractList(interimResult)
+	if err != nil {
+		return err
+	}
+	allItems = append(allItems, items...)
+
+	// Continue while there are more chunks.
+	for interimResult.GetContinue() != "" {
+		Continue(interimResult.GetContinue()).ApplyToList(&listOpts)
+		err = r.Get().
```

Comment:
I would re-arrange this code into a do/while structure so the actual API call doesn't have to be duplicated. Also this can use `c.List()` I think? That way we don't have that duplication as well.

```go
for {
  err := c.List(...)
  if err ...
  etc etc

  if result.GetContinue() == "" {
    break
  }
}
```

Reply:
Oh very neat!